packages/imperative/src/config/src/ProfileInfo.ts

```diff
@@ -527,6 +536,30 @@ export class ProfileInfo {
         return new Session(sessCfg);
     }
 
+    /**
+     * Checks if a profile (including nested subprofiles) exists in the given profiles object.
+     * Traverses recursively through the `profiles` property for nested profile support.
+     *
+     * @param profileName - The dotted path to the profile (ie: "base.sub1.sub2")
+     * @param profilesObj - The profiles object to search in.
+     * @returns true if the profile exists, false otherwise
+     */
+    private static profileExists(profileName: string, profilesObj: Record<string, IProfileNode>): boolean {
+        if (!profileName || !profilesObj) return false;
+
+        const levels = profileName.split(".");
+        let current: Record<string, IProfileNode> | IProfileNode = profilesObj;
+
+        for (const level of levels) {
+            if ((current as Record<string, IProfileNode>)[level]) {
+                current = (current as Record<string, IProfileNode>)[level].profiles || (current as Record<string, IProfileNode>)[level];
+            } else {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
Referring to my previous comment...

If you highlight over `layerProperties` in `ProfileInfo.mergeArgsForProfile`, you'll see that it it is of type `IConfig`:

<img width="747" height="72" alt="Image" src="https://github.com/user-attachments/assets/599ce8f6-8e52-4132-9531-91d40f149ef4" />

So, we can reuse the type of the `profiles` property in the `IConfig` interface here, with a couple code changes to avoid the repeated cast that's currently in place. Using the type recommended in the previous comment...

```suggestion
    private static profileExists(profileName: string, profilesObj: ProfileRecord): boolean {
        if (!profileName || !profilesObj) return false;

        const levels = profileName.split(".");
        let current: IConfigProfile | ProfileRecord = profilesObj;

        for (const level of levels) {
            const currentAsProfiles = current as ProfileRecord;
            if (currentAsProfiles[level]) {
                current = currentAsProfiles[level].profiles || currentAsProfiles[level];
            } else {
                return false;
            }
        }
        return true;
    }
```